src/RefreshToolbar/__tests__/RefreshToolbar.test.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { fireEvent, render } from '@testing-library/react';
+import Button from 'src/Button/Button';
+import RefreshToolbar from '../RefreshToolbar';
+
+describe('<RefreshToolbar />', () => {
+    let count = 0;
+    let refreshing = false;
+    const handleRefresh = () => {
+        refreshing = true;
+        count += 1;
+    };
+
+    it('renders button as correctly-styled icon', () => {
+        const { container } = render((
+            <RefreshToolbar
+                handleRefresh={handleRefresh}
+                isRefreshing={refreshing}
+                refreshProgressLabel="Loading..."
+                refreshTooltipText="Refresh LogSync File Details"
+            />
+        ));
+
+        expect(container).toMatchSnapshot();
+    });
+
+    it('renders tooltip on hover', () => {
+        const { container } = render((
+            <RefreshToolbar
+                handleRefresh={handleRefresh}
+                isRefreshing={refreshing}
+                refreshProgressLabel="Loading..."
+                refreshTooltipText="Refresh LogSync File Details"
+            />
+        ));
+
+        fireEvent.mouseOver(container.querySelector('button'));
```

Comment:
You may also be able to use react testing library's user-event library to test hover, which I think would basically be like `import userEvent from '@testing-library/user-event';` and then `userEvent.hover(container.querySelector('button'))` instead. The user-event library tries to act more how a user would interact with a page and is recommended over fireEvent when possible.

Also try avoiding querySelector in favor of getByRole (or getByText if there is no role with which to find) since query selectors can be null if nothing is found, but getBy will fail the test if not found. Query should only be used ideally when trying to assert the absence of an element on the page, such as `expect(container.querySelector('button').not.toBeInTheDocument();`

A cheat sheet for react testing library functions:
`queryBy`: lets you use a selector to find an element, avoid when possible. Can be null.
`getBy`: asserts the presence while finding an element, can't be null.
`await findBy`: asserts in combination with `waitFor` and will wait for the query to be present.

Finally, when using getBy and findBy, using it from the `screen` import is the best practice according to the creator since it reacts to changes in the render better.